auth0/resource_auth0_role.go

```diff
@@ -0,0 +1,167 @@
+package auth0
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"gopkg.in/auth0.v1"
+	"gopkg.in/auth0.v1/management"
+)
+
+func newRole() *schema.Resource {
+	return &schema.Resource{
+
+		Create: createRole,
+		Update: updateRole,
+		Read:   readRole,
+		Delete: deleteRole,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"role_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"user_ids": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+				ForceNew: true,
+			},
+			"permissions": {
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+						"resource_server_identifier": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+					},
+				},
+				Optional: true,
+			},
+		},
+	}
+}
+
+func createRole(d *schema.ResourceData, m interface{}) error {
+	c := buildRole(d)
+	api := m.(*management.Management)
+	if err := api.Role.Create(c); err != nil {
+		return err
+	}
+
+	users := buildUsers(d)
+	if err := api.Role.AssignUsers(*c.ID, users...); err != nil {
+		return err
+	}
+
+	permissions := buildPermissions(d)
+	if err := api.Role.AssignPermissions(*c.ID, permissions...); err != nil {
+		return err
+	}
+
+	d.SetId(auth0.StringValue(c.ID))
+	return readRole(d, m)
+}
+
+func readRole(d *schema.ResourceData, m interface{}) error {
+	api := m.(*management.Management)
+	c, err := api.Role.Read(d.Id())
+	if err != nil {
+		return err
+	}
+
+	d.SetId(auth0.StringValue(c.ID))
+	d.Set("role_id", c.ID)
+	d.Set("name", c.Name)
+	d.Set("description", c.Description)
+
+	users, err := api.Role.Users(d.Id())
+	if err != nil {
+		return err
+	}
+
+	userIDs := []string{}
+	for _, user := range users {
+		userIDs = append(userIDs, *user.ID)
+	}
+	d.Set("user_ids", userIDs)
```

Comment:
Please try to make use of the `List()` or `Set()` helpers as much as you can. You can see the connection resource for usage.